### PR TITLE
support auto generate for cumprod

### PR DESCRIPTION
**Pull request by Liyulingyue**
### PR types

Others 
### PR changes

Others 
### Describe

https://github.com/PaddlePaddle/Paddle/issues/51842 case45

### CODE REVIEW

File: paddle/phi/api/yaml/op_compat.yaml [hunk 1]
```diff
@@ -388,6 +388,13 @@
   outputs :
     out : Out
 
+- op : cumprod
+  backward : cumprod_grad
+  inputs :
+    {x : X, dim : Dim}
```
Comment (heavyrain-lzy) [inline paddle/phi/api/yaml/op_compat.yaml]: 这里的`dim`输入`attrs`，不要配置到inputs中，并且`dim`在`cumprod_op.cc`中参数名字也是`dim`(`AddAttr<int>(
        "dim",
        "（int), The dim along which the input tensors will be cumproded");`),所以这里不要配置`dim`

Comment (Liyulingyue) [reply paddle/phi/api/yaml/op_compat.yaml]: 收到~fixed

**Review by heavyrain-lzy — COMMENTED** [review]
LGTM 👍

**Comment by Liyulingyue** [comment]
@luotao1 cla报告卡住了，麻烦帮忙看一下该怎么处理~

**Review by luotao1 — APPROVED** [review]

**Pull request merged**